### ISSUE TITLE
Soot output naming conflict

### ISSUE THREAD

**Issue by Waldo1637** [Sable/soot#468]
If Soot attempts to write a jar file along with other outputs (ex: dumping call graphs), one of the outputs will fail because it attempts to create a jar file and a directory with the same name. 

For example, if the sootOutput directory is empty and you run Soot with options
```
-outjar -dump-cfg bb
```
the call graphs will be dumped but the following error occurs when trying to write the jar file:
```java
soot.CompilationDeathException: Cannot open output Jar file sootOutput\out.jar
	at soot.PackManager.setupJAR(PackManager.java:548)
	at soot.PackManager.writeOutput(PackManager.java:518)
	at soot.Main.run(Main.java:273)
	at soot.Main.main(Main.java:147)
```

In an alternative scenario, the sootOutput directory already contains a jar file named out.jar. Then when you run Soot, it will recreate the jar file but will fail to dump the CFGs with the following error:
```java
java.io.IOException: Unable to mkdirs sootOutput\out.jar\inst.test.Test1\int A0(int)
	at soot.util.PhaseDumper.makeDirectoryIfMissing(PhaseDumper.java:169)
	at soot.util.PhaseDumper.nextGraphFileName(PhaseDumper.java:192)
	at soot.util.PhaseDumper.dumpGraph(PhaseDumper.java:374)
	at soot.toolkits.graph.BriefUnitGraph.<init>(BriefUnitGraph.java:71)
	at soot.toolkits.graph.ZonedBlockGraph.<init>(ZonedBlockGraph.java:64)
	at soot.baf.toolkits.base.LoadStoreOptimizer$Instance.buildUnitToBlockMap(LoadStoreOptimizer.java:125)
	at soot.baf.toolkits.base.LoadStoreOptimizer$Instance.go(LoadStoreOptimizer.java:93)
	at soot.baf.toolkits.base.LoadStoreOptimizer.internalTransform(LoadStoreOptimizer.java:79)
	at soot.BodyTransformer.transform(BodyTransformer.java:51)
	at soot.Transform.apply(Transform.java:105)
	at soot.BodyPack.internalApply(BodyPack.java:49)
	at soot.Pack.apply(Pack.java:126)
	at soot.baf.BafBody.<init>(BafBody.java:122)
	at soot.baf.Baf.newBody(Baf.java:560)
	at soot.PackManager.convertJimpleBodyToBaf(PackManager.java:978)
	at soot.PackManager.runBodyPacks(PackManager.java:934)
	at soot.PackManager.runBodyPacks(PackManager.java:609)
	at soot.PackManager.runBodyPacks(PackManager.java:508)
	at soot.PackManager.runPacksNormally(PackManager.java:485)
	at soot.PackManager.runPacks(PackManager.java:392)
	at soot.Main.run(Main.java:271)
	at soot.Main.main(Main.java:147)
```

**Comment by Waldo1637** [comment]
Proposed fix is to create a new function getOutputJar() in soot.SourceLocator.java to return the full path to the jar file when Options.v().output_jar() is true. Should also modify getOutputDir() to only return the output directory (without the name of the jar file). Will also need to check all usages of the current function SourceLocator.v().getOutputDir() and make changes to use getOutputJar() as necessary.

**Comment by ericbodden** [comment]
Thanks a lot! Would you be able to provide a patch for us?

**Comment by StevenArzt** [comment]
It looks  as if this directly implements the proposed fix for issue #468. To me, it looks fine content-wise.
